portability-transfer/src/main/java/org/dataportabilityproject/transfer/PortabilityInMemoryDataCopier.java

```diff
@@ -121,16 +124,57 @@ private void copyHelper(
             exportAuthData,
             importAuthData,
             Optional.of(new ExportInformation(
-                continuationData.getPaginationData(), exportInformation.get().getContainerResource())));
+                continuationData.getPaginationData(),
+                exportInformation.get().getContainerResource())));
       }
 
       // Start processing sub-resources
       if (continuationData.getContainerResources() != null
           && !continuationData.getContainerResources().isEmpty()) {
         for (ContainerResource resource : continuationData.getContainerResources()) {
-          copyHelper(jobId, exportAuthData, importAuthData, Optional.of(new ExportInformation(null, resource)));
+          copyHelper(jobId, exportAuthData, importAuthData,
+              Optional.of(new ExportInformation(null, resource)));
+        }
+      }
+    }
+  }
+
+  private ExportResult export(UUID jobId, AuthData exportAuthData,
+      Optional<ExportInformation> exportInformation) {
+    ExportResult<?> exportResult = exporter.get().export(jobId, exportAuthData, exportInformation);
+    for (int attempts = 0; exportResult.getType() == ResultType.ERROR; attempts++) {
```

Comment:
can all this logic be moved into a helper method called validateResult() or retryIfNecessary() or something to that effect? I think that makes it more readable to not have the majority of the export() function be error handling

Reply:
I feel like it's simple enough to read now that I implemented your other suggestions.  What do you think?

Reply:
yep, looks good